src/SALib/analyze/delta.py

```diff
@@ -58,26 +58,27 @@ def analyze(problem, X, Y, num_resamples=10,
         raise RuntimeError("Confidence level must be between 0-1.")
 
     # equal frequency partition
-    M = min(np.ceil(N ** (2 / (7 + np.tanh((1500 - N) / 500)))), 48)
+    M = min(int(np.ceil(N ** (2 / (7 + np.tanh((1500 - N) / 500))))), 48)
     m = np.linspace(0, N, M + 1)
     Ygrid = np.linspace(np.min(Y), np.max(Y), 100)
 
     keys = ('delta', 'delta_conf', 'S1', 'S1_conf')
-    S = ResultDict((k, np.zeros(D)) for k in keys)
+    S = ResultDict((k, np.empty(D)) for k in keys)

```

Comment:
Do note that using `np.empty` can cause issues if the array is not subsequently filled with values. This caused some hard to find bugs in the past.

Reply:
Yes, I am aware of this.

That said, the implementations here should all subsequently fill in the arrays and use of `empty` has a slight performance benefit.

If `D` values were not filled, I think that's sign of a larger issue.
As it stands though I'm happy to revert everything back to using `np.zeros` although it will pain me to do so...

Reply:
Okay.